File: rb/features.rb
# GuildWars2 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module GuildWars2Features
  def self.make_feature(name)
    case name
    when "base"
      GuildWars2BaseFeature.new
    when "test"
      GuildWars2TestFeature.new
    else
      GuildWars2BaseFeature.new
    end
  end
end

File: py/features.py
# GuildWars2 SDK feature factory

from feature.base_feature import GuildWars2BaseFeature
from feature.test_feature import GuildWars2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: GuildWars2BaseFeature(),
        "test": lambda: GuildWars2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
